tools/Makefile

```diff
@@ -84,19 +46,39 @@ VPATH := $(filter-out -I,$(INCLUDES))
 .PHONY: all allopt opt.opt # allopt and opt.opt are synonyms
 allopt: opt.opt
 
+programs_byte := \
+  ocamldep ocamlprof ocamlcp ocamloptp ocamlmklib  \
+  ocamlmktop ocamlcmt dumpobj ocamlobjinfo \
+  primreq stripdebug cmpbyt
+install_files += $(filter $(installed_tools), $(programs_byte))
+programs_opt := $(programs_byte:%=%.opt)
+
+programs_byte_exe := $(programs_byte:%=%$(EXE))
+programs_opt_exe := $(programs_opt:%=%$(EXE))
+$(programs_byte) $(programs_opt): %: $(if $(EXE), %$(EXE), )
+
+$(programs_byte_exe):
+	$(CAMLC) $(LINKFLAGS) -I $(ROOTDIR) -o $@ $(filter-out %.cmi,$^)
+
+$(programs_opt_exe):
+	$(CAMLOPT_CMD) $(LINKFLAGS) -I $(ROOTDIR) -o $@ $(filter-out %.cmi,$^)
+
+all: $(programs_byte_exe)
+opt.opt: $(programs_opt_exe)
+clean::
+	rm -f $(programs_byte_exe) $(programs_opt_exe)
+
 # The dependency generator
 
 CAMLDEP_OBJ=ocamldep.cmo
 CAMLDEP_IMPORTS= \
   $(ROOTDIR)/compilerlibs/ocamlcommon.cma \
   $(ROOTDIR)/compilerlibs/ocamlbytecomp.cma
-ocamldep$(EXE): LINKFLAGS += -compat-32
-$(call byte_and_opt,ocamldep,$(CAMLDEP_IMPORTS) $(CAMLDEP_OBJ),)
-ocamldep$(EXE): depend.cmi
-ocamldep.opt$(EXE): depend.cmi
 
-clean::
-	rm -f ocamldep ocamldep.exe ocamldep.opt ocamldep.opt.exe
+ocamldep$(EXE): LINKFLAGS += -compat-32
+ocamldep$(EXE): depend.cmi $(CAMLDEP_IMPORTS) $(CAMLDEP_OBJ)
```

Comment:
You included a `filter-out` in the rule, so I think you can put `depend.cmi` in `CAMLDEP_IMPORTS` (or `ocamldep_objects`)

Reply:
Done